Assets/LeapMotion/Core/Scripts/Utils/Utils.cs

```diff
@@ -734,6 +734,85 @@ public struct ChildrenEnumerator : IEnumerator<Transform> {
 
     #endregion
 
+    #region Component Utils
+
+    /// <summary>
+    /// Recursively searches the hierarchy of the argument Transform to find all of the
+    /// Components of type ComponentType (the first type argument) that should be "owned"
+    /// by the OwnerType component type (the second type argument).
+    /// 
+    /// If a child GameObject itself has an OwnerType component, that
+    /// child is ignored, and its children are ignored -- the assumption being that such
+    /// a child owns itself and any ComponentType components beneath it.
+    /// 
+    /// For example, a call to FindOwnedChildComponents with ComponentType Collider and
+    /// OwnerType Rigidbody would return all of the Colliders that are attached to the
+    /// rootObj Rigidbody, but none of the colliders that are attached to a rootObj's
+    /// child's own Rigidbody.
+    /// 
+    /// Optionally, ComponentType components of inactive GameObjects can be included
+    /// in the returned list; by default, these components are skipped.
+    /// 
+    /// This is not a cheap method to call, but it does not allocate garbage, so it is safe
+    /// for use at runtime.
+    /// </summary>
+    /// 
+    /// <typeparam name="ComponentType">
+    /// The component type to search for.
+    /// </typeparam>
+    /// 
+    /// <typeparam name="OwnerType">
+    /// The component type that assumes ownership of any ComponentType in its own Transform
+    /// or its Transform's children/grandchildren.
+    /// </typeparam>
+    public static void FindOwnedChildComponents<ComponentType, OwnerType>
+                                               (OwnerType rootObj,
+                                                List<ComponentType> ownedComponents,
+                                                bool includeInactiveObjects = false)
+                                               where OwnerType : Component {
+      ownedComponents.Clear();
+      Stack<Transform> toVisit = Pool<Stack<Transform>>.Spawn();
+      List<ComponentType> componentsBuffer = Pool<List<ComponentType>>.Spawn();
+
+      try {
+        // Traverse the hierarchy of this object's transform to find
+        // all of its Colliders.
+        toVisit.Push(rootObj.transform);
+        Transform curTransform;
+        while (toVisit.Count > 0) {
+          curTransform = toVisit.Pop();
+
+          // Recursively search children and children's children.
+          foreach (var child in curTransform.GetChildren()) {
+            // Ignore children with OwnerType components of their own; its own OwnerType
+            // component owns its own ComponentType components and the ComponentType
+            // components of its children.
+            if (child.GetComponent<OwnerType>() == null
+                && (includeInactiveObjects || child.gameObject.activeSelf)) {
```

Comment:
Not sure if this is what you intend.  Active self can be true even if the game object is deactivated due to a parent being inactive.  Maybe activeHierarchy is what you want instead?

Reply:
yeah, that's fair. I could see both being utilized but I think activeInHierarchy is the better default; we can add another option later if we need it